CART/CARTLib/core/DataUnitBase.py

```diff
@@ -33,8 +34,11 @@ def __init__(
         # Resource tracking
         self.resources = {}
         self.uid = case_data.get("uid", None)
-        self.validated = False
-        self._validate()
+        # HACK: Just work around so that we dont call the validate method before setup all required resources
+        # in the subclasses
+        if do_validation:
+            self.validated = False
+            self._validate()
```

Comment:
Potential "attribute not defined" error
```suggestion
        self.validated = False
        if do_validation:
            self._validate()
```